src/irmin-pack/inode.ml

```diff
@@ -776,41 +783,37 @@ module Make_ext
                with type hash = H.t
                 and type Val.metadata = Node.metadata
                 and type Val.step = Node.step)
-    (Pack : Pack.S with type value = Inter.Elt.t and type key = H.t) =
+    (P : Pack.MAKER with type key = H.t and type index = Pack_index.Make(H).t) =
 struct
   module Key = H
+  module Pack = P.Make (Inter.Elt)
 
   type 'a t = 'a Pack.t
   type key = Key.t
   type value = Inter.Val.t
+  type index = Pack.index
 
   let mem t k = Pack.mem t k
-
-  let unsafe_find ~check_integrity t k =
-    match Pack.unsafe_find ~check_integrity t k with
-    | None -> None
-    | Some v ->
-        let v = Inter.Val_impl.of_bin v in
-        Some v
+  let unsafe_find ~check_integrity t k = Pack.unsafe_find ~check_integrity t k
```

Comment:
if the only usage of `unsafe_find` is in `find` maybe you can move this directly there?